tests/integration/internal/test_lock.py

```diff
@@ -74,3 +74,12 @@ def test_release():
     assert user1_lock._has_lock() is False
     assert user2_lock._acquire_lock()
     assert user1_lock._acquire_lock() is False
+
+    user2_lock.release()
+
+
+def test_nested_context_managers():
+    with Lock(".foo", max_age=datetime.timedelta(seconds=2)) as user1_lock:
+        with Lock(".foo", max_age=datetime.timedelta(seconds=2)) as user2_lock:
```

Comment:
add comment describing the test. mention blocking behavior of __enter__() when used as a context manager
add documentation on lock itself.